apps/mark/backend/src/app.ts

```diff
@@ -159,6 +182,64 @@ export function buildApi(
       logger,
       store: workspace.store.getUiStringsStore(),
     }),
+
+    incrementBallotsPrintedCount() {
+      store.setBallotsPrintedCount(store.getBallotsPrintedCount() + 1);
+    },
+
+    async setPollsState(input: { pollsState: PollsState }) {
+      const newPollsState = input.pollsState;
+      const oldPollsState = store.getPollsState();
+
+      store.setPollsState(newPollsState);
+
+      assert(newPollsState !== 'polls_closed_initial');
+      const logEvent = (() => {
+        switch (newPollsState) {
+          case 'polls_closed_final':
+            return LogEventId.PollsClosed;
+          case 'polls_paused':
+            if (oldPollsState === 'polls_closed_final') {
+              // logging case handled by ResetPollsToPausedButton
+              return undefined;
+            }
+            return LogEventId.VotingPaused;
+          case 'polls_open':
+            if (oldPollsState === 'polls_closed_initial') {
+              return LogEventId.PollsOpened;
+            }
+            return LogEventId.VotingResumed;
+          /* istanbul ignore next */
+          default:
+            throwIllegalValue(newPollsState);
+        }
+      })();
+      if (logEvent) {
+        await logger.log(logEvent, 'poll_worker', { disposition: 'success' });
+      }
+    },
+
+    setTestMode(input: { isTestMode: boolean }) {
+      store.setTestMode(input.isTestMode);
+      store.setPollsState('polls_closed_initial');
+      store.setBallotsPrintedCount(0);
+    },
+
+    setPrecinctSelection(input: {
+      precinctSelection: PrecinctSelection;
+    }): void {
+      store.setPrecinctSelection(input.precinctSelection);
+      store.setBallotsPrintedCount(0);
+    },
+
+    getElectionState(): ElectionState {
+      return {
+        precinctSelection: store.getPrecinctSelection(),
+        ballotsPrintedCount: store.getBallotsPrintedCount(),
+        isTestMode: store.getTestMode(),
+        pollsState: store.getPollsState(),
```

Comment:
I lean toward just leaving the logic separate for now. It's what we do in `scan`, and since it's a read from a table with one row I'm not very concerned about the performance.

Reply:
Makes sense